components/clarinet-sdk/src/unit_tests.rs

```diff
@@ -0,0 +1,279 @@
+
+        let (execution, _) =
+            match session.invoke_contract_call(contract, method, &args, sender, test_name.into()) {
+                Ok(res) => res,
+                Err(diagnostics) => {
+                    let mut message = format!(
+                        "{}: {}::{}({})",
+                        "Contract call runtime error",
+                        contract,
+                        method,
+                        args.join(", ")
+                    );
+                    if let Some(diag) = diagnostics.last() {
+                        message = format!("{} -> {}", message, diag.message);
+                    }
+                    log!("message: {}", message);
+                    std::process::exit(1);
+                }
+            };
+
+        let result = match execution.result {
+            EvaluationResult::Snippet(result) => utils::to_raw_value(&result.result),
+            _ => unreachable!("Contract value from snippet"),
+        };
+        let events = js_sys::Array::new_with_length(execution.events.len() as u32);
+        for (i, event) in execution.events.iter().enumerate() {
+            events.set(i as u32, encode_to_js(&serialize_event(event)).unwrap())
+        }
+
+        CallContractRes { result, events }
+    }
+
+    #[wasm_bindgen(js_name=callReadOnlyFn)]
+    pub fn call_read_only_fn(
+        &mut self,
+        contract: &str,
+        method: &str,
+        js_args: Vec<js_sys::Uint8Array>,
+        sender: &str,
+    ) -> Result<CallContractRes, String> {
+        let interface = self.get_function_interface(contract, method)?;
+        if interface.access != ContractInterfaceFunctionAccess::read_only {
+            return Err(format!("{method} is not a read-only function"));
+        }
+
+        Ok(self.invoke_contract_call(contract, method, &js_args, sender, "read-only call"))
+    }
+
+    #[wasm_bindgen(js_name = "callPublicFn")]
+    pub fn call_public_fn(
+        &mut self,
+        contract: &str,
+        method: &str,
+        js_args: Vec<js_sys::Uint8Array>,
+        sender: &str,
+    ) -> Result<CallContractRes, String> {
+        let interface = self.get_function_interface(contract, method)?;
+        if interface.access != ContractInterfaceFunctionAccess::public {
+            return Err(format!("{method} is not a public function"));
+        }
+
+        let res = self.invoke_contract_call(contract, method, &js_args, sender, "public call");
+
+        let session = self.session.as_mut().unwrap();
+        session.advance_chain_tip(1);
+
+        Ok(res)
+    }
+
+    // #[wasm_bindgen(js_name=mineBlock)]
+    // pub fn mine_block(&mut self, js_params: JsValue) -> JsValue {
+    //     // @todo
+    //     // re-implement chainhooks logic
+    //     // https://github.com/hirosystems/clarinet/blob/c6d65974f489606c0a8c57a6d0f01cfb993eb79b/components/clarinet-cli/src/runner/api_v1.rs#L718-L811
```

Comment:
We can get rid of this feature.

Reply:
Do you mean `mine_block` or the chainhook logic @lgalabru?

Reply:
Chainhook logic.

Reply:
We'll still need a `mine_block` like approach, but I was wondering if we should keep the same struct, or use the types that we're using in https://github.com/hirosystems/clarinet/blob/develop/components/stacks-devnet-js/lib/index.ts.
Any take?